sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientRetryPolicy.java

```diff
@@ -152,6 +160,33 @@ public Mono<ShouldRetryResult> shouldRetry(Exception e) {
         return this.throttlingRetry.shouldRetry(e);
     }
 
+      private boolean gatewayRequestCanFailoverOnTimeout(RxDocumentServiceRequest request, CosmosException clientException) {
+        //Query Plan requests
+        if(request.getResourceType() == ResourceType.Document
+            && request.getOperationType() == OperationType.QueryPlan) {
+            return true;
+        }
+
+        //Meta data request check
+        boolean isMetaDataRequest = (request.getOperationType() != OperationType.ExecuteJavaScript
+            && request.getResourceType() == ResourceType.StoredProcedure)
+            || request.getResourceType() != ResourceType.Document;
+
+        //Meta Data Read
+        if(isMetaDataRequest && request.isReadOnly()) {
+              return true;
+        }
+
+        //Data Plane Read
```

Comment:
Please also create a tracking working item in Java side to follow up the behavior of data plane writes